Parse exceptions when using Azure OpenAI only

Using the latest main branch, I am seeing parse exceptions when using Azure OpenAI. When using native OpenAI I don't see these errors. 

I sometimes get a partial response, like this:

```
You:
> Name the us presidents of the 20th century
✅
..Server:
The Presidents of the United States during the 20th century were:

1. William McKinley (1897–1901)
2. Theodore Roosevelt (1901–1909)
3. William Howard Taft (1909–1913)
4. Woodrow Wilson (1913–1921)
5. Warren G. Harding (1921–1923)
6. Calvin Coolidge (1923–192
```

But then output stops and the logs are spammed with this:

```
20:40:05.297 [pool-4-thread-1] INFO  a.l.kafka.runner.KafkaReaderWrapper -- Received 1 records from Kafka topics [output-topic-0]
20:40:05.301 [pool-4-thread-1] INFO  a.l.kafka.runner.KafkaReaderWrapper -- Received 1 records from Kafka topics [output-topic-0]
20:40:05.300 [pipeline-document-to-json-1-runner-8ce7d553-5282-424e-a013-2311bd01b426] ERROR a.l.r.agent.CompositeAgentProcessor -- Internal Error processing record: KafkaRecord.KafkaConsumerRecord(record=ConsumerRecord(topic = input-topic, partition = 0, leaderEpoch = 0, offset = 0, CreateTime = 1695069601951, serialized key size = -1, serialized value size = 42, headers = RecordHeaders(headers = [RecordHeader(key = langstream-client-session-id, value = [66, 56, 65, 54, 50, 67, 55, 68, 45, 49, 55, 55, 52, 45, 52, 53, 55, 66, 45, 56, 48, 53, 65, 45, 68, 57, 48, 52, 57, 50, 69, 48, 54, 48, 50, 51])], isReadOnly = false), key = null, value = Name the us presidents of the 20th century))
reactor.core.Exceptions$ReactiveException: com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 0)): has to be escaped using backslash to be included in name
 at [Source: (String)"{"id":"chatcmpl-80FKER2i7WcyHzoo1PSMUGIlbHo7G","object":"chat.completion.chunk","created":1695069602,"model":"gpt-4","choices":[{"index":0,"finish_reason":null,"delta":{"content":"9"},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"se\u0000\u0000\u0000\u0000\u0000\u0000xual":\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"[truncated 4009 chars]; line: 1, column: 308] (through reference chain: com.azure.ai.openai.models.ChatCompletions["choices"]->java.util.ArrayList[0]->com.azure.ai.openai.models.ChatChoice["content_filter_results"])
	at reactor.core.Exceptions.propagate(Exceptions.java:408)
	at reactor.core.publisher.BlockingIterable$SubscriberIterator.hasNext(BlockingIterable.java:169)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at ai.langstream.ai.agents.services.impl.OpenAICompletionService.getChatCompletions(OpenAICompletionService.java:91)
	at com.datastax.oss.streaming.ai.ChatCompletionsStep.processAsync(ChatCompletionsStep.java:149)
	at ai.langstream.ai.agents.GenAIToolKitAgent.processStep(GenAIToolKitAgent.java:138)
	at ai.langstream.ai.agents.GenAIToolKitAgent.processRecord(GenAIToolKitAgent.java:116)
	at ai.langstream.ai.agents.GenAIToolKitAgent.process(GenAIToolKitAgent.java:92)
	at ai.langstream.api.runner.code.AgentCodeAndLoader$3.lambda$process$0(AgentCodeAndLoader.java:208)
	at ai.langstream.api.runner.code.AgentCodeAndLoader.executeNoExceptionWithContextClassloader(AgentCodeAndLoader.java:56)
	at ai.langstream.api.runner.code.AgentCodeAndLoader$3.process(AgentCodeAndLoader.java:207)
	at ai.langstream.runtime.agent.CompositeAgentProcessor.invokeProcessor(CompositeAgentProcessor.java:165)
	at ai.langstream.runtime.agent.CompositeAgentProcessor.lambda$invokeProcessor$0(CompositeAgentProcessor.java:202)
	at ai.langstream.api.runner.code.SingleRecordAgentProcessor.process(SingleRecordAgentProcessor.java:40)
	at ai.langstream.api.runner.code.AgentCodeAndLoader$3.lambda$process$0(AgentCodeAndLoader.java:208)
	at ai.langstream.api.runner.code.AgentCodeAndLoader.executeNoExceptionWithContextClassloader(AgentCodeAndLoader.java:56)
	at ai.langstream.api.runner.code.AgentCodeAndLoader$3.process(AgentCodeAndLoader.java:207)
	at ai.langstream.runtime.agent.CompositeAgentProcessor.invokeProcessor(CompositeAgentProcessor.java:165)
	at ai.langstream.runtime.agent.CompositeAgentProcessor.process(CompositeAgentProcessor.java:225)
	at ai.langstream.api.runner.code.AgentCodeAndLoader$3.lambda$process$0(AgentCodeAndLoader.java:208)
	at ai.langstream.api.runner.code.AgentCodeAndLoader.executeNoExceptionWithContextClassloader(AgentCodeAndLoader.java:56)
	at ai.langstream.api.runner.code.AgentCodeAndLoader$3.process(AgentCodeAndLoader.java:207)
	at ai.langstream.runtime.agent.AgentRunner.runProcessorAgent(AgentRunner.java:832)
	at ai.langstream.runtime.agent.AgentRunner.runMainLoop(AgentRunner.java:644)
	at ai.langstream.runtime.agent.AgentRunner.runJavaAgent(AgentRunner.java:412)
	at ai.langstream.runtime.agent.AgentRunner.run(AgentRunner.java:190)
	at ai.langstream.runtime.agent.AgentRunner.runAgent(AgentRunner.java:129)
	at ai.langstream.runtime.tester.LocalApplicationRunner.lambda$executeAgentRunners$2(LocalApplicationRunner.java:208)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 0)): has to be escaped using backslash to be included in name
 at [Source: (String)"{"id":"chatcmpl-80FKER2i7WcyHzoo1PSMUGIlbHo7G","object":"chat.completion.chunk","created":1695069602,"model":"gpt-4","choices":[{"index":0,"finish_reason":null,"delta":{"content":"9"},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"se\u0000\u0000\u0000\u0000\u0000\u0000xual":\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"[truncated 4009 chars]; line: 1, column: 308] (through reference chain: com.azure.ai.openai.models.ChatCompletions["choices"]->java.util.ArrayList[0]->com.azure.ai.openai.models.ChatChoice["content_filter_results"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:402)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:361)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1826)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:280)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:463)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:352)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:359)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:542)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:564)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:439)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:352)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4730)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3677)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3645)
	at com.azure.ai.openai.implementation.OpenAIServerSentEvents.handleCurrentLine(OpenAIServerSentEvents.java:98)
	at com.azure.ai.openai.implementation.OpenAIServerSentEvents.lambda$mapByteBuffersToEvents$0(OpenAIServerSentEvents.java:58)
	at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.onNext(FluxConcatMapNoPrefetch.java:183)
	at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.runAsync(FluxPublishOn.java:440)
	at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.run(FluxPublishOn.java:527)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	... 3 common frames omitted
```

--- a/langstream-agents/langstream-ai-agents/src/main/java/ai/langstream/ai/agents/services/impl/OpenAICompletionService.java
+++ b/langstream-agents/langstream-ai-agents/src/main/java/ai/langstream/ai/agents/services/impl/OpenAICompletionService.java
@@ -88,13 +88,14 @@ public class OpenAICompletionService implements CompletionsService {
                     new ChatCompletionsConsumer(
                             streamingChunksConsumer, minChunksPerMessage, finished);
 
-            flux.subscribe(chatCompletionsConsumer);
-
             flux.doOnError(
-                    error -> {
-                        log.error("Internal error while processing the streaming response", error);
-                        finished.completeExceptionally(error);
-                    });
+                            error -> {
+                                log.error(
+                                        "Internal error while processing the streaming response",
+                                        error);
+                                finished.completeExceptionally(error);
+                            })
+                    .subscribe(chatCompletionsConsumer);
 
             return finished.thenApply(
                     ___ -> {

--- a/langstream-agents/langstream-ai-agents/src/main/java/ai/langstream/ai/agents/services/impl/OpenAICompletionService.java
+++ b/langstream-agents/langstream-ai-agents/src/main/java/ai/langstream/ai/agents/services/impl/OpenAICompletionService.java
@@ -95,7 +95,8 @@ public class OpenAICompletionService implements CompletionsService {
                                         error);
                                 finished.completeExceptionally(error);
                             })
-                    .subscribe(chatCompletionsConsumer);
+                    .doOnNext(chatCompletionsConsumer)
+                    .subscribe();
 
             return finished.thenApply(
                     ___ -> {
@@ -109,8 +110,7 @@ public class OpenAICompletionService implements CompletionsService {
         } else {
             com.azure.ai.openai.models.ChatCompletions chatCompletions =
                     client.getChatCompletions((String) options.get("model"), chatCompletionsOptions)
-                            .toFuture()
-                            .get();
+                            .block();
             result.setChoices(
                     chatCompletions.getChoices().stream()
                             .map(c -> new ChatChoice(convertMessage(c)))

--- a/langstream-agents/langstream-ai-agents/src/main/java/ai/langstream/ai/agents/services/impl/OpenAICompletionService.java
+++ b/langstream-agents/langstream-ai-agents/src/main/java/ai/langstream/ai/agents/services/impl/OpenAICompletionService.java
@@ -35,7 +35,6 @@ import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
 import reactor.core.publisher.Flux;
@@ -84,11 +83,19 @@ public class OpenAICompletionService implements CompletionsService {
             Flux<com.azure.ai.openai.models.ChatCompletions> flux =
                     client.getChatCompletionsStream(
                             (String) options.get("model"), chatCompletionsOptions);
-            Stream<com.azure.ai.openai.models.ChatCompletions> model = flux.toStream();
+
             ChatCompletionsConsumer chatCompletionsConsumer =
                     new ChatCompletionsConsumer(
                             streamingChunksConsumer, minChunksPerMessage, finished);
-            model.forEach(chatCompletionsConsumer);
+
+            flux.subscribe(chatCompletionsConsumer);
+
+            flux.doOnError(
+                    error -> {
+                        log.error("Internal error while processing the streaming response", error);
+                        finished.completeExceptionally(error);
+                    });
+
             return finished.thenApply(
                     ___ -> {
                         result.setChoices(

--- a/langstream-runtime/langstream-runtime-tester/pom.xml
+++ b/langstream-runtime/langstream-runtime-tester/pom.xml
@@ -47,6 +47,10 @@
           <groupId>${project.groupId}</groupId>
           <artifactId>langstream-pulsar-runtime</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>${project.groupId}</groupId>
+          <artifactId>langstream-codestorage-azure-blob-storage</artifactId>
+        </exclusion>
       </exclusions>
     </dependency>
     <dependency>
@@ -54,6 +58,12 @@
       <artifactId>langstream-webservice</artifactId>
       <version>${project.version}</version>
       <classifier>original</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>${project.groupId}</groupId>
+          <artifactId>langstream-codestorage-azure-blob-storage</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>${project.groupId}</groupId>

